### PR TITLE
MES-2716  - Rest of  Cat C Analytics

### PR DESCRIPTION
**Pull request by TomBillingtonUK**
## Description
- Fixed issue with Reversing Diagram Modal Analytics not firing
- Fixed issue with Non-Pass Completion Page Analytics not firing
- Added Length Changed Analytics Event on the Reversing Diagram Modal for B+E & C
- Added Width Changed Analytics Event on the Reversing Diagram Modal for B+E & C
- Added Conducted Language Analytics Event on Pass Completion 2 Page for all categories.

## Checklist

- [x] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by Cullima — APPROVED** [review]

File: src/pages/pass-finalisation/__tests__/pass-finalisation.analytics.effects.spec.ts [hunk 1]
```diff
@@ -246,7 +248,7 @@ describe('Pass Finalisation Analytics Effects', () => {
         done();
       });
     });
-    it('should call not call logEvent if there is no activty code', (done) => {
+    it('should call not call logEvent if there is no activity code', (done) => {
```
Comment (mgoldson) [inline src/pages/pass-finalisation/__tests__/pass-finalisation.analytics.effects.spec.ts]: Looks like an extra 'call' here?

File: src/pages/pass-finalisation/__tests__/pass-finalisation.analytics.effects.spec.ts [hunk 2]
```diff
@@ -300,4 +302,72 @@ describe('Pass Finalisation Analytics Effects', () => {
...
+        expect(analyticsProviderMock.logEvent)
+          .toHaveBeenCalledWith(
+            AnalyticsEventCategories.POST_TEST,
+            AnalyticsEvents.LANGUAGE_CHANGED,
+            Language.ENGLISH,
+          );
+        done();
+      });
+    });
+    it('should call not call logEvent if there is no activity code', (done) => {
```
Comment (mgoldson) [inline src/pages/pass-finalisation/__tests__/pass-finalisation.analytics.effects.spec.ts]: Looks like an extra 'call' here?

File: src/pages/pass-finalisation/__tests__/pass-finalisation.analytics.effects.spec.ts [hunk 3]
```diff
@@ -300,4 +302,72 @@ describe('Pass Finalisation Analytics Effects', () => {
...
+        expect(analyticsProviderMock.logEvent)
+          .toHaveBeenCalledWith(
+            AnalyticsEventCategories.POST_TEST,
+            AnalyticsEvents.LANGUAGE_CHANGED,
+            Language.CYMRAEG,
+          );
+        done();
+      });
+    });
+    it('should call not call logEvent if there is no activity code', (done) => {
```
Comment (mgoldson) [inline src/pages/pass-finalisation/__tests__/pass-finalisation.analytics.effects.spec.ts]: Looks like an extra 'call' here?

File: src/pages/non-pass-finalisation/__tests__/non-pass-finalisation.analytics.effects.spec.ts [hunk 4]
```diff
@@ -120,5 +128,166 @@ describe('Non Pass Finalisation Analytics Effects', () => {
...
+        expect(analyticsProviderMock.logEvent)
+          .toHaveBeenCalledWith(
+            AnalyticsEventCategories.POST_TEST,
+            AnalyticsEvents.LANGUAGE_CHANGED,
+            Language.CYMRAEG,
+          );
+        done();
+      });
+    });
+    it('should call not call logEvent if there is no activity code', (done) => {
```
Comment (mgoldson) [inline src/pages/non-pass-finalisation/__tests__/non-pass-finalisation.analytics.effects.spec.ts]: Looks like an extra 'call' here?

File: src/pages/non-pass-finalisation/__tests__/non-pass-finalisation.analytics.effects.spec.ts [hunk 5]
```diff
@@ -120,5 +128,166 @@ describe('Non Pass Finalisation Analytics Effects', () => {
...
+        expect(analyticsProviderMock.logEvent)
+          .toHaveBeenCalledWith(
+            AnalyticsEventCategories.POST_TEST,
+            AnalyticsEvents.LANGUAGE_CHANGED,
+            Language.ENGLISH,
+          );
+        done();
+      });
+    });
+    it('should call not call logEvent if there is no activity code', (done) => {
```
Comment (mgoldson) [inline src/pages/non-pass-finalisation/__tests__/non-pass-finalisation.analytics.effects.spec.ts]: Looks like an extra 'call' here?

**Review by mgoldson — APPROVED** [review]
Nice work Tom! Question on a few test descriptions but nothing to stop this going in.

**Pull request merged**